packages/smooth_app/lib/pages/product/product_attributes_page.dart

```diff
@@ -0,0 +1,344 @@
+        }
+        allTerms.add(term);
+      });
+    });
+
+    allTerms = allTerms.toSet().toList();
+
+    final AppLocalizations appLocalizations = AppLocalizations.of(context);
+    return _ListTile(
+      allTerms: allTerms,
+      leading: const SvgIcon('assets/cacheTintable/scale-balance.svg',
+          dontAddColor: true),
+      title: appLocalizations.nutrition_page_title,
+      hasTrailing: true,
+      onTap: () async {
+        if (!await ProductRefresher().checkIfLoggedIn(
+          context,
+          isLoggedInMandatory: true,
+        )) {
+          return;
+        }
+        AnalyticsHelper.trackProductEdit(
+          AnalyticsEditEvents.nutrition_Facts,
+          widget.product.barcode!,
+        );
+
+        if (!mounted) {
+          return;
+        }
+
+        await NutritionPageLoaded.showNutritionPage(
+          product: widget.product,
+          isLoggedInMandatory: true,
+          context: context,
+        );
+      },
+    );
+  }
+}
+
+class Ingredients extends StatefulWidget {
+  const Ingredients({super.key, required this.product});
+
+  final Product product;
+
+  @override
+  State<Ingredients> createState() => _IngredientsState();
+}
+
+class _IngredientsState extends State<Ingredients> {
+  List<String> allIngredients = <String>[];
+  @override
+  Widget build(BuildContext context) {
+    widget.product.ingredients?.forEach((Ingredient element) {
+      if (element.text != null) {
+        allIngredients.add(element.text!);
+      }
+    });
+    final AppLocalizations appLocalizations = AppLocalizations.of(context);
+
+    return _ListTile(
+      allTerms: allIngredients,
+      leading: const SvgIcon('assets/cacheTintable/ingredients.svg',
+          dontAddColor: true),
+      title: appLocalizations.ingredients,
+      onTap: () async => ProductFieldOcrIngredientEditor().edit(
+        context: context,
+        product: widget.product,
+      ),
+    );
+  }
+}
+
+class AttributeItems extends StatefulWidget {
+  const AttributeItems({required this.helper, required this.product});
+  final AbstractSimpleInputPageHelper helper;
+  final Product product;
+
+  @override
+  State<AttributeItems> createState() => _AttributeItemsState();
+}
+
+class _AttributeItemsState extends State<AttributeItems> {
+  late final List<String> _localTerms;
+
+  @override
+  void initState() {
+    super.initState();
+    widget.helper.reInit(widget.product);
+    _localTerms = List<String>.of(widget.helper.terms);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final AppLocalizations appLocalizations = AppLocalizations.of(context);
+    return _ListTile(
+      allTerms: _localTerms,
+      leading: widget.helper.getIcon(),
+      title: widget.helper.getTitle(appLocalizations),
+      onTap: () async => ProductFieldSimpleEditor(widget.helper).edit(
+        context: context,
+        product: widget.product,
+      ),
+    );
+  }
+}
+
+class _ListTile extends StatefulWidget {
```

Comment:
Please put it in a specific file, and name it with a less vague name, like `AttributeFirstRowWidget`.